### PR TITLE
[SDL 0173] Add customDataTYpe parameter and add  VEHICLEDATA_OEM_CUSTOM_DATA enum value

### PR DESCRIPTION
**Pull request by mkorniichuk**
This PR is **ready** for review.

Related to https://github.com/smartdevicelink/sdl_evolution/pull/802 
### Summary
Current PR implements custom data type in response to mobile for Subscribe/Unsubscribe VehicleData

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/vehicle_info_plugin/include/vehicle_info_plugin/custom_vehicle_data_manager_impl.h [hunk 1]
```diff
@@ -32,6 +32,9 @@ class CustomVehicleDataManagerImpl : public CustomVehicleDataManager {
   virtual smart_objects::SmartObject CreateHMIMessageParams(
       const std::set<std::string>& item_names) OVERRIDE;
 
+  std::string GetVehicleDataItemType(
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/vehicle_info_plugin/include/vehicle_info_plugin/custom_vehicle_data_manager_impl.h]: @mkorniichuk  suppose, this also can be const

**Comment by atiwari9** [comment]
Approved

**Comment by LuxoftAKutsan** [comment]
@jacobkeeler  this PR is related to https://github.com/smartdevicelink/sdl_evolution/issues/806
Should I merge it to feature branch or you will review it separately?

**Comment by LuxoftAKutsan** [comment]
Merging to feature branch after accepting revision https://github.com/smartdevicelink/sdl_evolution/issues/806

**Pull request merged**